### PR TITLE
Java doc examples for Kafka, #151

### PR DESCRIPTION
**Pull request by patriknw**
* use jdbc in Kafka doc examples, also for Scala
* fixed some missing javadsl AP
* doc examples

References #151

### CODE REVIEW

File: docs/src/main/paradox/kafka.md [hunk 1]
```diff
@@ -6,11 +6,12 @@ A typical source for Projections is messages from Kafka. Akka Projections has in
...
 
-Akka Projections can store the offsets from Kafka in a @ref:[relational DB with Slick](slick.md).
+Akka Projections can store the offsets from Kafka in a @ref:[relational DB with JDBC](jdbc.md)
+or in @ref:[relational DB with Slick](slick.md).
 
-The `SlickProjection` envelope handler returns a `DBIO` that will be run by the projection. This means that the target database
-operations can be run in the same transaction as the storage of the offset, which means that @ref:[exactly-once](slick.md#exactly-once)
-processing semantics is supported. It also offers @ref:[at-least-once](slick.md#at-least-once) semantics.
+The `JdbcProjection` @scala[or `SlickProjection`] envelope handler will be run by the projection. This means that the target database
+operations can be run in the same transaction as the storage of the offset, which means that @ref:[exactly-once](jdbc.md#exactly-once)
```
Comment (renatocaval) [inline docs/src/main/paradox/kafka.md:13]: May also link to slick docs?

File: examples/src/test/java/jdocs/kafka/KafkaDocExample.java [hunk 2]
```diff
@@ -4,9 +4,213 @@
...
+    private final ProjectionId projectionId;
+
+    private final Map<String, Integer> state = new HashMap<>();
+
+    public WordCountHandler(ProjectionId projectionId) {
+      this.projectionId = projectionId;
+    }
+
+    @Override
+    public void process(HibernateJdbcSession session, ConsumerRecord<String, String> envelope) {
```
Comment (renatocaval) [inline examples/src/test/java/jdocs/kafka/KafkaDocExample.java:67]: I guess the goal here is to make it a default handler (ie: `CompletionState<Done>`) and demo at-least-once. Correct?

File: examples/src/test/java/jdocs/kafka/KafkaDocExample.java [hunk 3]
```diff
@@ -4,9 +4,213 @@
...
+    private final String topic;
+    private final SendProducer<String, String> sendProducer;
+
+    public WordPublisher(String topic, SendProducer<String, String> sendProducer) {
+      this.topic = topic;
+      this.sendProducer = sendProducer;
+    }
+
+    @Override
+    public void process(HibernateJdbcSession session, WordEnvelope envelope) throws Exception {
```
Comment (renatocaval) [inline examples/src/test/java/jdocs/kafka/KafkaDocExample.java:129]: same here, at-least-once with async handler. No?

**Comment by patriknw** [comment]
You want to wait because you think we will not have jdbc ready when we publish the docs by end of next week? I’d rather take that risk than having to rework this again when it has been stale.

**Comment by renatocaval** [comment]
I trust we will finish it, but we will rework it anyway because of support for at-least-once.

File: examples/src/test/java/jdocs/kafka/KafkaDocExample.java (hunk 3 above)
Comment (patriknw) [reply examples/src/test/java/jdocs/kafka/KafkaDocExample.java:129]: yes, publishing with this handler should at-least-once, and there is a FIXME for that below

File: examples/src/test/java/jdocs/kafka/KafkaDocExample.java (hunk 2 above)
Comment (patriknw) [reply examples/src/test/java/jdocs/kafka/KafkaDocExample.java:67]: No, consuming should be exactlyOnce. I have created issue https://github.com/akka/akka-projection/issues/256 for making the example more interesting.

File: docs/src/main/paradox/kafka.md (hunk 1 above)
Comment (patriknw) [reply docs/src/main/paradox/kafka.md:13]: Slick is mentioned at the top of the page, but otherwise we should promote JDBC mostly.

**Comment by patriknw** [comment]
I'll go ahead and merging this.

**Pull request merged**